Neos.Flow.Log/Classes/Backend/JsonFileBackend.php

```diff
@@ -0,0 +1,66 @@
+<?php
+namespace Neos\Flow\Log\Backend;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+/**
+ * A log backend which writes log entries into a file in an easy to parse json format
+ *
+ * @api
+ */
+class JsonFileBackend extends FileBackend
+{
+    /**
+     * @param string $message The message to log
+     * @param integer $severity One of the LOG_* constants
+     * @param mixed $additionalData A variable containing more information about the event to be logged
+     * @param string $packageKey Key of the package triggering the log (determined automatically if not specified)
+     * @param string $className Name of the class triggering the log (determined automatically if not specified)
+     * @param string $methodName Name of the method triggering the log (determined automatically if not specified)
+     * @return void
+     */
+    public function append($message, $severity = LOG_INFO, $additionalData = null, $packageKey = null, $className = null, $methodName = null)
+    {
+
+        if ($severity > $this->severityThreshold) {
+            return;
+        }
+
+        if (function_exists('posix_getpid')) {
+            $processId = posix_getpid();
+        } else {
+            $processId = 0;
+        }
+
+        $remoteIp = ($this->logIpAddress === true) ? (isset($_SERVER['REMOTE_ADDR']) ? $_SERVER['REMOTE_ADDR'] : '') : '';
+        $severityLabel = (isset($this->severityLabels[$severity])) ? strtolower(trim($this->severityLabels[$severity])) : 'unknown';
+
+        $logEntryData = [
+            'timestamp' => date(\DateTime::ATOM),
+            'processId' => $processId,
+            'severity' => $severityLabel,
+            'remoteIp' => $remoteIp,
+            'message' => $message,
+            'origin' => [
+                'packageKey' => $packageKey,
+                'className' => $className,
+                'methodName' => $methodName
+            ],
+            'additionalData' => $additionalData
```

Comment:
What happens is additionalData are not JSON encodable ?

And maybe an array_filter to avoid null value, lots of value are null by default, so the JSON will contains a lots of key without values ?

Reply:
> What happens is additionalData are not JSON encodable ?
Good point. Any idea, how to cover this? Testing the encoding before and change the array to an error if it fails? Or build something like https://github.com/neos/flow-development-collection/blob/b84273d1728bc7591ce233b37434aa502b4219c5/Neos.Flow.Log/Classes/PlainTextFormatter.php to sanitize the array?

> And maybe an array_filter to avoid null value
I am not sure about that, If you have the same fields in all you messages, then you could rely on that fact when parsing. My use case is ingesting the log messages into elasticsearch and there we have the empty fields anyway

Reply:
Let `json_encode` handle that and just look for `json_last_error` if it returns `false`.

Reply:
technically whoever is logging should take care. Putting that data into a log file also means serialisation of sorts (worst case toString), so I think it's fine to expect the data to be json_encode-able